src/decorator/CircuitBreakerDecorator.test.ts

```diff
@@ -0,0 +1,42 @@
+import { CircuitBreakerError } from '../CircuitBreakerError';
+import { CircuitBreakerDecorator } from './CircuitBreakerDecorator';
+
+describe('CircuitBreakerDecorator test Suite', () => {
+    @CircuitBreakerDecorator({
+        failureThreshold: 1,
+        recoveryTimeout: 3000,
+    })
+    class ClassUnderTest {
+        public unstableFn(shouldThrow: boolean): boolean {
```

Comment:
Maybe we could have two functions (unstable & stable) and test that with a failure threshold set to 3, 2 calls to unstable function then 1 call to stable function, then 1 call to unstable function does not open the circuit.

Basically the test I'd write would be "check that all functions share the same circuit breaker state"